pkg/systemd/overview-cards/usageCard.jsx

```diff
@@ -29,59 +29,91 @@ import "./usageCard.less";
 
 const _ = cockpit.gettext;
 
-const UPDATE_DELAY = 5000;
+const METRICS_SPEC = {
+    payload: "metrics1",
+    source: "internal",
+    interval: 5000,
+    metrics: [
+        { name: "cpu.basic.user", derive: "rate" },
+        { name: "cpu.basic.system", derive: "rate" },
+        { name: "cpu.basic.nice", derive: "rate" },
+        { name: "memory.used" },
+    ]
+};
 
 export class UsageCard extends React.Component {
     constructor(props) {
         super(props);
 
-        this.state = { pollingEnabled: true };
-        this.updateMemoryInfo = this.updateMemoryInfo.bind(this);
-    }
+        this.metrics_channel = null;
 
-    componentDidMount() {
-        this.updateMemoryInfo();
+        this.state = { memTotal: 0, memUsed: 0, numCpu: 1, cpuUsed: 0 };
 
-        cockpit.addEventListener("visibilitychange", () => {
-            this.setState((prevState, _) => ({ pollingEnabled: !prevState.pollingEnabled }));
-        }, () => this.updateMemoryInfo());
-    }
+        machine_info.cpu_ram_info().done(info => this.setState({
+            memTotal: Number((info.memory / (1024 * 1024 * 1024)).toFixed(1)),
+            numCpu: info.cpus,
+        }));
+
+        this.onVisibilityChange = this.onVisibilityChange.bind(this);
+        this.onMetricsUpdate = this.onMetricsUpdate.bind(this);
 
-    componentWillUnmount() {
-        this.setState({ pollingEnabled: false });
+        cockpit.addEventListener("visibilitychange", this.onVisibilityChange);
+        this.onVisibilityChange();
     }
 
-    updateMemoryInfo() {
-        if (!this.state.pollingEnabled)
+    onVisibilityChange() {
+        if (cockpit.hidden && this.metrics_channel !== null) {
+            this.metrics_channel.removeEventListener("message", this.onMetricsUpdate);
+            this.metrics_channel.close();
+            this.metrics_channel = null;
             return;
+        }
 
-        machine_info.cpu_ram_info().done(info => {
-            this.setState({
-                memTotal: Number((info.memory / (1024 * 1024 * 1024)).toFixed(1)),
-                memAvailable: Number((info.available_memory / (1024 * 1024 * 1024)).toFixed(1))
-            });
-        });
-        window.setTimeout(this.updateMemoryInfo.bind(this), UPDATE_DELAY);
+        if (!cockpit.hidden && this.metrics_channel === null) {
+            this.metrics_channel = cockpit.channel(METRICS_SPEC);
+            this.metrics_channel.addEventListener("closed", (ev, error) => console.error("metrics closed:", error));
+            this.metrics_channel.addEventListener("message", this.onMetricsUpdate);
+        }
+    }
+
+    onMetricsUpdate(event, message) {
+        const data = JSON.parse(message);
```

Comment:
You need to handle sample compression here, see doc/protocol.md.  See storaged/client.js instance_sampler for an example.

Reply:
You did that in the FIXUP, I just reformatted it to use ES6 syntax. Thanks!